force-app/main/default/classes/OmdbConnection.cls

```diff
@@ -0,0 +1,138 @@
+global with sharing class OmdbConnection {
+    private String httpMethod = 'GET';
+    private String omdbURLBase = 'http://www.omdbapi.com/';
+
+    // (?) TODO:    -> Reference from encrypted .txt file (?)
```

Comment:
You generally want to remove TODOs before you commit. They can be confusing for someone pulling down the code and not knowing if the TODOs are still relevant or not. They are okay to have in code while you are working, but once you finish a feature, you should remove them

Reply:
Ok that's good to know thank you! Generally speaking, would it be best practice to keep the 'key' in a separate file? I'm sure it's fine for this project, but I can always change that if needed.

Reply:
It really depends... There are a number of ways that Salesforce allows for storing secrets and encrypted information each with pros and cons depending on the value you are trying to store and how secure you need to keep it.

In this case we probably would want to store the API key and URL in a Custom Metadata or Custom Setting within Salesforce. The data isn't really a secret per se, so we don't have to worry too much about keeping the value hidden like say a username or password. Storing the value in a Custom Metadata or Custom Setting would just allow for us to maintain the values without needing to update a class each time in the event it changes